test/list/reduce_single_item.wren

```diff
@@ -0,0 +1 @@
+[1].reduce {|a, b| a } // expect runtime error: Range start out of bounds.
```

Comment:
My expectation would be that this returns 1 instead of throwing a runtime error. I think it should only throw an error if the list is empty and the single arity `reduce` is called.